### ISSUE TITLE
jax.numpy.i0 drops the second derivative at zero

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40627]
### Description

`jax.numpy.i0` appears to return an incorrect zero second derivative at zero.

For the modified Bessel function I0:

```
I0(x) = 1 + x**2 / 4 + O(x**4)
```

so:

```
I0''(0) = 1/2
```

In the example below, the differentiated function is `i0(2 * t - 1)`. By the chain rule, the second derivative with respect to `t` at `t = 0.5` should be `2.0`. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "")
os.environ.setdefault("OMP_NUM_THREADS", "1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

def target(t):
    return jnp.i0(2 * t - 1)

x = jnp.asarray(0.5, dtype=jnp.float64)
expected = 2.0

actual = float(jax.grad(jax.grad(target))(x))

print("actual:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-12):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
actual: 0.0
expected: 2.0
```

### Expected result

```
2.0
```

The forward value is correct, but the second derivative is not. The expected value is finite, normal, and exactly representable in `float64`. JAX returns `0.0`, suggesting that the higher-order autodiff rule for `jax.numpy.i0` loses the curvature at zero.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
Thanks for the report. Comparing to #40628, this looks like it's an issue both for `jax.scipy.special.i0` and for `jax.numpy.i0`, though they fail in different ways:
```python
>>> import jax
>>> jax.grad(jax.grad(jax.scipy.special.i0))(0.0)
Array(1.5000001, dtype=float32, weak_type=True)
>>> jax.grad(jax.grad(jax.numpy.i0))(0.0)
Array(0., dtype=float32, weak_type=True)
```
I believe the first is due to `grad(abs)` returning `1.0` at `x = 0`, and this propagates through to result in an incorrect overall gradient. The second is due to the fact that the custom JVP rule returns a constant 0.0 at x=0: https://github.com/jax-ml/jax/blob/8c1e217afb2b17f60a47474ee8c37e947405eaa8/jax/_src/numpy/lax_numpy.py#L6156

I'll think about how to adjust this custom JVP rule so that it will result in correct higher-order gradients.
